### PR TITLE
add group id for replicator logs

### PR DESCRIPTION
**Pull request by qiujiayu**
在 Replicator 的日志中增加 group id 信息，方便定位多Raft group 下的问题。

### CODE REVIEW

File: jraft-core/src/main/java/com/alipay/sofa/jraft/core/Replicator.java [hunk 1]
```diff
@@ -610,26 +614,27 @@ private void startHeartbeatTimer(final long startMs) {
...
-            LOG.error("Fail to check install snapshot connection to peer={}, give up to send install snapshot request.", this.options.getPeerId().getEndpoint());
+            LOG.error("Fail to check install snapshot connection to peer={}, give up to send install snapshot request, group id {}.",
+                    this.options.getPeerId().getEndpoint(), this.options.getGroupId());
             block(Utils.nowMs(), RaftError.EHOSTDOWN.getNumber());
             return;
         }
         try {
             Requires.requireTrue(this.reader == null,
-                "Replicator %s already has a snapshot reader, current state is %s", this.options.getPeerId(),
+                "Replicator %s already has a snapshot reader, current state is %s", this.toString(),
```
Comment (fengjiachun) [inline jraft-core/src/main/java/com/alipay/sofa/jraft/core/Replicator.java:637]: Perhaps `(peerId , groupId, replicatorType)` are enough, the information in `this.toString` is too complicated.

File: jraft-core/src/main/java/com/alipay/sofa/jraft/core/Replicator.java [hunk 2]
```diff
@@ -909,7 +914,7 @@ public static ThreadId start(final ReplicatorOptions opts, final RaftOptions raf
         r.id = new ThreadId(r, r);
         r.id.lock();
         notifyReplicatorStatusListener(r, ReplicatorEvent.CREATED);
-        LOG.info("Replicator={}@{} is started", r.id, r.options.getPeerId());
+        LOG.info("Replicator={} is started", r);
```
Comment (fengjiachun) [inline jraft-core/src/main/java/com/alipay/sofa/jraft/core/Replicator.java:917]: Perhaps `(peerId , groupId, replicatorType)` are enough.

File: jraft-core/src/main/java/com/alipay/sofa/jraft/core/Replicator.java [hunk 3]
```diff
@@ -932,7 +937,7 @@ public static void waitForCaughtUp(final String groupId, final ThreadId id, fina
         }
         try {
             if (r.catchUpClosure != null) {
-                LOG.error("Previous wait_for_caught_up is not over");
+                LOG.error("Previous wait_for_caught_up is not over, peer:{}, group id:{}", r.options.getPeerId(), r.options.getGroupId());
```
Comment (fengjiachun) [inline jraft-core/src/main/java/com/alipay/sofa/jraft/core/Replicator.java:940]:
```suggestion
                LOG.error("Previous wait_for_caught_up is not over, peer: {}, group id: {}", r.options.getPeerId(), r.options.getGroupId());
```

File: jraft-core/src/main/java/com/alipay/sofa/jraft/core/Replicator.java [hunk 4]
```diff
@@ -1510,7 +1516,8 @@ private static boolean onAppendEntriesReturned(final ThreadId id, final Inflight
         if (response.getTerm() != r.options.getTerm()) {
             r.resetInflights();
             r.setState(State.Probe);
-            LOG.error("Fail, response term {} dismatch, expect term {}", response.getTerm(), r.options.getTerm());
+            LOG.error("Fail, response term {} dismatch, expect term {}, peer {}, group id{}", response.getTerm(),
```
Comment (fengjiachun) [inline jraft-core/src/main/java/com/alipay/sofa/jraft/core/Replicator.java:1519]:
```suggestion
            LOG.error("Fail, response term {} dismatch, expect term {}, peer {}, group {}", response.getTerm(),
```

File: jraft-core/src/main/java/com/alipay/sofa/jraft/core/Replicator.java [hunk 5]
```diff
@@ -718,7 +723,7 @@ static boolean onInstallSnapshotReturned(final ThreadId id, final Replicator r,
                 LOG.info(sb.toString());
                 notifyReplicatorStatusListener(r, ReplicatorEvent.ERROR, status);
                 if ((r.consecutiveErrorTimes++) % 10 == 0) {
-                    LOG.warn("Fail to install snapshot at peer={}, error={}", r.options.getPeerId(), status);
+                    LOG.warn("Fail to install snapshot at peer={}, error={}, group id = {}", r.options.getPeerId(), status, r.options.getGroupId());
```
Comment (fengjiachun) [inline jraft-core/src/main/java/com/alipay/sofa/jraft/core/Replicator.java:726]:
```suggestion
                    LOG.warn("Fail to install snapshot at peer={}, error={}, group id={}", r.options.getPeerId(), status, r.options.getGroupId());
```

**Comment by fengjiachun** [comment]
测试失败了？

**Comment by qiujiayu** [comment]
> 测试失败了？

  已解决

**Review by fengjiachun — APPROVED** [review]

**Pull request merged**